ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveUtils.java

```diff
@@ -107,6 +109,7 @@ public static String escapeString(String str) {
   static final byte[] tabEscapeBytes = "\\t".getBytes();;
   static final byte[] tabUnescapeBytes = "\t".getBytes();
   static final byte[] ctrlABytes = "\u0001".getBytes();
+  static final Pattern BRANCH = Pattern.compile("branch_(.*)");
```

Comment:
If table name or database name has branch? Will this work?

Reply:
It is only used to extract branch name from the table suffix `tableMetaRef `https://github.com/apache/hive/pull/4292/files#diff-42b2d7dcd39e3f43dd8da948858388f92d2213a4963ab7e08cb92df6cc06453dR40

e.g. the three layer table ref identifier `dbName.tblName.tblMetaRef` --> `testdb.testtbl.branch_test1`, we will get actual branch from the `tblMetaRef` ->`test1`.

So it does't matter if table or database has branch.

Reply:
Cool